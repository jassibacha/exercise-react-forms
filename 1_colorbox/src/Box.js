import React from 'react';
import './Box.css';

const Box = ({ width, height, backgroundColor, id, deleteBox }) => {
    return (
        <div
            className="Box"
            style={{
                width: width + 'px',
                height: height + 'px',
                backgroundColor: backgroundColor,
            }}
            key={id}
        >
            <button className="delete" onClick={deleteBox}>
                X
            </button>
        </div>
    );
};

export default Box;
