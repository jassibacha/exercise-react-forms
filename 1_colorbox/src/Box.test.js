import { render, screen } from '@testing-library/react';
import Box from './Box';

it('renders without crashing', function () {
    render(
        <Box
            width="100"
            height="100"
            backgroundColor="#FF0000"
            id="test-id"
            deleteBox={() => {}}
        />
    );
});

it('matches snapshot', function () {
    const { asFragment } = render(
        <Box
            width="100"
            height="100"
            backgroundColor="#FF0000"
            id="test-id"
            deleteBox={() => {}}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});
