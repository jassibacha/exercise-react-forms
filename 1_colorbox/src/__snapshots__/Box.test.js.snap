// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="width: 100px; height: 100px; background-color: rgb(255, 0, 0);"
  >
    <button
      class="delete"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
